### DISCUSSION TITLE
How to handle key value store creation?

### DISCUSSION THREAD

**Discussion by sergeda** [nats-io/nats.rs#1318] [Q&A] [answered]
How to create key value store only if it doesn't exist?
There is no error like `BucketExist`:
```
pub enum CreateKeyValueErrorKind {
    InvalidStoreName,
    TooLongHistory,
    JetStream,
    BucketCreate,
    TimedOut,
}
```
Then what will happen when we call `create_key_value` and specify name of the bucket that already exist? What type of error will be return in such case?

**Comment by Jarema** [comment, marked as answer]
There will be no error if the underlying stream already exists with the same configuraiton.
If you want to catch `stream name already in use with a different configuration"` you need to downcast the `JetStream` error.

I think it's a good idea to expose this one directly. 
Will add it.
